### PR TITLE
Add files example

### PR DESCRIPTION
**Pull request by tchssk**
This example shows how to build an API that serves static assets. It includes how to use custom `FileSystem` (related to https://github.com/goadesign/goa/pull/1296).

### CODE REVIEW

**Comment by raphael** [comment]
Could you maybe as a little bit of details in the README around go-bin-data, what it does and hour it's used? Great example :)

**Comment by tchssk** [comment]
I tried adding explanation about go-bindata. c0c4ffa

File: files/README.md [hunk 1]
```diff
@@ -15,3 +15,28 @@ Then compile and run the service:
 go build
 ./files
 ```
+
+## Resources
+
+This example contains following two resources.
```
Comment (raphael) [inline files/README.md]: ...contains the following two resources:

File: files/README.md [hunk 2]
```diff
@@ -15,3 +15,28 @@ Then compile and run the service:
...
+
+* schema
+* swagger
+
+### Schema
+
+Schema is served from native file system so you have to deploy the binary with `public/schema` directory and keep their path relationship.
```
Comment (raphael) [inline files/README.md]: The `schema` resource contains a single endpoint which serves the static files contained in the `public/schema` directory. This means the directory must exist and its relative path must be `public/schema`.

File: files/README.md [hunk 3]
```diff
@@ -15,3 +15,28 @@ Then compile and run the service:
...
+
+### Swagger
+
+Swagger is served from file system of [https://github.com/elazarl/go-bindata-assetfs](github.com/elazarl/go-bindata-assetfs).
```
Comment (raphael) [inline files/README.md]: The `swagger` resource also contains a single endpoint however it leverages [https://github.com/elazarl/go-bindata-assetfs](github.com/elazarl/go-bindata-assetfs) to serve assets embedded within the compiled binary.

File: files/README.md [hunk 4]
```diff
@@ -15,3 +15,28 @@ Then compile and run the service:
...
+
+* [go-bindata](https://github.com/jteeuwen/go-bindata) is
+
+    > A small utility which generates Go code from any file. Useful for embedding binary data in a Go program.
+
+* [go-bindata-assetfs](https://github.com/elazarl/go-bindata-assetfs)
+
+    > Serves embedded files from `jteeuwen/go-bindata` with `net/http`.
+
+When you use these packages, the binary contains static assets. It means you don't need to deploy `public/swagger` directory.
```
Comment (raphael) [inline files/README.md]: Using these packages alleviates the need for deploying the `public/swagger` directory.

**Comment by tchssk** [comment]
Thank you for your suggestions. 

38866a5 improves the document.

**Comment by raphael** [comment]
Great thanks!

**Pull request merged**